source/MaterialXTest/MaterialXGenGlsl/GenGlsl.cpp

```diff
@@ -140,6 +147,14 @@ static void generateGlslCode()
     const mx::GenOptions genOptions;
     mx::FilePath optionsFilePath = testRootPath / mx::FilePath("_options.mtlx");
     tester.validate(genOptions, optionsFilePath);
+
+    // Test glsl 400 with uniform layout qualifier extensions
+    const mx::FilePath glsl42LogPath("genglsl_glsl420_layout_generate_test.txt");
+    writeShadersToDisk = true;
+    GlslShaderGeneratorTester testerglsl42(mx::GlslShaderGenerator::create(), testRootPaths, libSearchPath, srcSearchPath, glsl42LogPath, writeShadersToDisk);
+    
+    // enable tester to use userdata
+    testerglsl42.validate(genOptions, optionsFilePath, true);
```

Comment:
Maybe separate his out to two different tests, so we can choose to run one or the other or both. And have them both not write all files to disc by default.

Reply:
Sounds good I'll put a separate PR for this.